File: typescript/src/amen/resources/webhooks.ts
import type { AmenClient } from "../client.js";
import type { Webhook } from "../types.js";
export class Webhooks {
  constructor(private c: AmenClient) {}
  list() { return this.c.request<Webhook[]>("GET", "/web-hooks/"); }
  /** `secret_key` in the response is shown ONLY now — store it in a secret manager immediately. */
  create(url: string) { return this.c.request<Webhook>("POST", "/web-hooks/", { json: { url } }); }
  delete(id: string) { return this.c.request<void>("DELETE", `/web-hooks/${id}`); }
}

File: python/src/amen/resources/webhooks.py
from __future__ import annotations
from ..models import Webhook


class Webhooks:
    def __init__(self, client): self._c = client
    def list(self) -> list[Webhook]:
        return [Webhook.from_api(w) for w in self._c.request("GET", "/web-hooks/")]
    def create(self, url: str) -> Webhook:
        """The returned secret_key is shown ONLY now — store it in a secret manager immediately."""
        return Webhook.from_api(self._c.request("POST", "/web-hooks/", json={"url": url}))
    def delete(self, webhook_id: str) -> None:
        self._c.request("DELETE", f"/web-hooks/{webhook_id}")
